ocrd_network/ocrd_network/processing_worker.py

```diff
@@ -0,0 +1,327 @@
+"""
+Abstraction for the Processing Server unit in this arch:
+https://user-images.githubusercontent.com/7795705/203554094-62ce135a-b367-49ba-9960-ffe1b7d39b2c.jpg
+
+Calls to native OCR-D processor should happen through
+the Processing Worker wrapper to hide low level details.
+According to the current requirements, each ProcessingWorker
+is a single OCR-D Processor instance.
+"""
+
+import json
+import logging
+from os import environ, getpid
+import requests
+from typing import Any, List
+
+import pika.spec
+import pika.adapters.blocking_connection
+
+from ocrd import Resolver
+from ocrd_utils import getLogger
+from ocrd.processor.helpers import run_cli, run_processor
+from ocrd_network.database import (
+    sync_initiate_database,
+    sync_set_processing_job_state
+)
+from ocrd_network.models.job import StateEnum
+from ocrd_network.rabbitmq_utils import (
+    OcrdProcessingMessage,
+    OcrdResultMessage,
+    RMQConsumer,
+    RMQPublisher
+)
+from ocrd_network.utils import (
+    verify_database_uri,
+    verify_and_parse_mq_uri
+)
+
+try:
+    # This env variable must be set before importing from Keras
+    environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+    from tensorflow.keras.utils import disable_interactive_logging
+    # Enabled interactive logging throws an exception
+    # due to a call of sys.stdout.flush()
+    disable_interactive_logging()
+except Exception:
+    # Nothing should be handled here if TF is not available
+    pass
+
+
+class ProcessingWorker:
+    def __init__(self, rabbitmq_addr, mongodb_addr, processor_name, ocrd_tool: dict, processor_class=None) -> None:
+        self.log = getLogger(__name__)
+        # TODO: Provide more flexibility for configuring file logging (i.e. via ENV variables)
+        file_handler = logging.FileHandler(f'/tmp/worker_{processor_name}_{getpid()}.log', mode='a')
+        logging_format = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+        file_handler.setFormatter(logging.Formatter(logging_format))
+        file_handler.setLevel(logging.DEBUG)
+        self.log.addHandler(file_handler)
+        sync_initiate_database(mongodb_addr)  # Database client
+
+        try:
+            self.db_url = verify_database_uri(mongodb_addr)
```

Comment:
Why do we need to introduce a new variable `db_url` here? If it's just for the log on the next line, can't we re-use the `mongodb_addr`?